iota-execution/latest/iota-verifier/src/entry_points_verifier.rs

```diff
@@ -180,7 +181,17 @@ fn verify_entry_function_impl(
 
     let all_non_ctx_params = match params.0.last() {
```

Comment:
I would suggest simplified version of these checks.
1. Removing inner match
2. Less manual indexing
3. A bit more structured 

```suggestion
    let all_non_ctx_params = match params.0.last() {
        Some(last_param) if TxContext::kind(view, last_param) != TxContextKind::None => {
            let mut non_ctx_params = &params.0[..params.0.len() - 1];

            if let Some((second_last, rest)) = non_ctx_params.split_last() {
                if AuthContext::kind(view, second_last) != AuthContextKind::None {
                    non_ctx_params = rest;
                }
            }

            non_ctx_params
        }
        _ => &params.0[..],
    };
```

Reply:
[ddecdf4](https://github.com/iotaledger/iota/pull/9368/commits/ddecdf4a2aed9dab398084eb81e49edee5f833ed)